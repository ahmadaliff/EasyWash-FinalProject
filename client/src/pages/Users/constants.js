export const GET_USERS = 'users/GET_USERS';
export const GET_UNVERIFIED_USERS = 'users/GET_UNVERIFIED_USERS';
export const VERIFY_USER = 'users/VERIFY_USER';
export const DELETE_USER = 'users/DELETE_USER';
export const SET_USERS = 'users/SET_USERS';
export const RESET_USERS = 'users/RESET_USERS';
