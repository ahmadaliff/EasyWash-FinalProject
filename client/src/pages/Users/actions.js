import { DELETE_USER, GET_USERS, RESET_USERS, SET_USERS } from './constants';

export const actionGetUsers = (search, limit, page) => ({
  type: GET_USERS,
  search,
  limit,
  page,
});

export const actionSetUsers = (users) => ({
  type: SET_USERS,
  users,
});

export const actionDeleteUser = (data) => ({
  type: DELETE_USER,
  data,
});

export const actionResetUsers = () => ({
  type: RESET_USERS,
});
