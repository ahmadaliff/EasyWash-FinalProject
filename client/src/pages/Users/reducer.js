import { produce } from 'immer';
import { DELETE_USER, RESET_USERS, SET_USERS } from './constants';

export const initialState = {
  users: null,
};

export const storedKey = ['users'];

export const usersReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case SET_USERS:
        draft.users = action.users;
        break;
      case DELETE_USER: {
        const filteredDataUsers = state.users.data.filter((val) => val.id !== action.data);

        const tempObj = { ...state.users };
        tempObj.data = filteredDataUsers;
        tempObj.totalRows -= 1;

        draft.users = tempObj;
        break;
      }
      case RESET_USERS:
        return initialState;
    }
  });

export default usersReducer;
