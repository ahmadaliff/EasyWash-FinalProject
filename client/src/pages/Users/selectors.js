import { createSelector } from 'reselect';
import { initialState } from '@pages/Users/reducer';

const selectUsersState = (state) => state.users || initialState;

export const selectUsers = createSelector(selectUsersState, (state) => state.users);
