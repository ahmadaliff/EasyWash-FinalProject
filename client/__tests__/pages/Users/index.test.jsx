import Users from '@pages/Users';
import { render, fireEvent, act } from '@utils/testHelper';

let wrapper;
const mockNavigate = jest.fn();
const mockDispatch = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockDispatch,
}));

jest.mock('@components/MapLeaflet', () => jest.fn().mockImplementation(() => <div />));

jest.mock('@pages/Users/selectors', () => ({
  selectUsers: () => ({
    data: [
      {
        status: 'app_pickUp',
        userId: 1,
        totalRows: 1,
        id: 1,
        totalPrice: 20000,
      },
    ],
    totalRows: 1,
  }),
}));

jest.mock('@components/DetailOrder', () => jest.fn().mockImplementation(() => <p data-testid="dialog-detail-order" />));

beforeEach(() => {
  act(() => {
    wrapper = render(<Users />);
  });
});

describe('Users Page', () => {
  test('render', () => {
    const { getByTestId } = wrapper;
    expect(getByTestId('users-page')).toBeInTheDocument();
  });
  test('navigate back when back button click', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('back-button');
    await act(async () => {
      fireEvent.click(button);
    });
    expect(mockNavigate).toHaveBeenCalledWith(-1);
  });
  test('change tab users unverified', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-unverified-user');
    act(() => {
      fireEvent.click(button);
    });
    expect(mockDispatch).toHaveBeenCalled();
  });
  test('shouldbe clicked buttton verify', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-unverified-user');
    act(() => {
      fireEvent.click(button);
    });
    const buttonVerify = getByTestId('button-verify');
    await act(async () => {
      fireEvent.click(buttonVerify);
    });
  });
  test('shouldbe clicked buttton decline', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-unverified-user');
    act(() => {
      fireEvent.click(button);
    });
    const buttonDecline = getByTestId('button-decline');
    await act(async () => {
      fireEvent.click(buttonDecline);
    });
  });
  test('change tab users verified', () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-verified-users');
    fireEvent.click(button);
    expect(mockDispatch).toHaveBeenCalled();
  });
  test('shouldbe clicked buttton delete', async () => {
    const { getByTestId } = wrapper;
    const button = getByTestId('button-delete');
    await act(async () => {
      fireEvent.click(button);
    });
  });
  test('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
