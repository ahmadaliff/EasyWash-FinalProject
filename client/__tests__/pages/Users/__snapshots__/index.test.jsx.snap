// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users Page Matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="mainWrap"
        data-testid="users-page"
      >
        <div
          class="header"
        >
          <div
            class="rowOne"
          >
            <h3>
              Manajemen User
            </h3>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
              data-testid="back-button"
              tabindex="0"
              type="button"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowBackIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
              Kembali
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="rowTwo"
          >
            <div
              class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth toggleRole css-mgs850-MuiToggleButtonGroup-root"
              role="group"
            >
              <button
                aria-pressed="false"
                class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU kRQhAE MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
                data-testid="button-verified-users"
                tabindex="0"
                type="button"
                value="false"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="VerifiedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m23 12-2.44-2.79.34-3.69-3.61-.82-1.89-3.2L12 2.96 8.6 1.5 6.71 4.69 3.1 5.5l.34 3.7L1 12l2.44 2.79-.34 3.7 3.61.82L8.6 22.5l3.4-1.47 3.4 1.46 1.89-3.19 3.61-.82-.34-3.69L23 12zm-12.91 4.72-3.8-3.81 1.48-1.48 2.32 2.33 5.85-5.87 1.48 1.48-7.33 7.35z"
                  />
                </svg>
                 
                Terverifikasi
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
              <button
                aria-pressed="true"
                class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU kRQhAE MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
                data-testid="button-unverified-user"
                tabindex="0"
                type="button"
                value="true"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="UnpublishedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M21.19 21.19 2.81 2.81 1.39 4.22l2.27 2.27C2.61 8.07 2 9.96 2 12c0 5.52 4.48 10 10 10 2.04 0 3.93-.61 5.51-1.66l2.27 2.27 1.41-1.42zm-10.6-4.59-4.24-4.24 1.41-1.41 2.83 2.83.18-.18 1.41 1.41-1.59 1.59zm3-5.84-7.1-7.1C8.07 2.61 9.96 2 12 2c5.52 0 10 4.48 10 10 0 2.04-.61 3.93-1.66 5.51L15 12.17l2.65-2.65-1.41-1.41-2.65 2.65z"
                  />
                </svg>
                 
                Tidak Terverifikasi
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <div
              class="searchInputWrap"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSearch css-i4bv87-MuiSvgIcon-root"
                data-testid="SearchIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                />
              </svg>
              <input
                class="searchInput"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
        >
          <div
            class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
          >
            <table
              aria-label="sticky table"
              class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
            >
              <thead
                class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                >
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead false css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Nama Lengkap
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead email css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Email
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead false css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Peran
                  </th>
                  <th
                    class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                    scope="col"
                  >
                    Aksi
                  </th>
                </tr>
              </thead>
              <tbody
                class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              >
                <tr
                  class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
                >
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody false css-1ex1afd-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody email css-1ex1afd-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody false css-1ex1afd-MuiTableCell-root"
                  />
                  <td
                    class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                  >
                    <div
                      class="tableAction"
                    >
                      <button
                        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttonAction css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                        data-testid="button-delete"
                        tabindex="0"
                        type="button"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                          data-testid="DeleteIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                          />
                        </svg>
                        <div
                          class="email"
                        >
                          Hapus Akun
                        </div>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </td>
                </tr>
              </tbody>
            </table>
          </div>
          <div
            class="MuiTablePagination-root tablePagination css-jtlhu6-MuiTablePagination-root"
          >
            <div
              class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
            >
              <div
                class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
              />
              <p
                class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
                id=":r14:"
              >
                Rows per page:
              </p>
              <div
                class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
              >
                <div
                  aria-controls=":r15:"
                  aria-expanded="false"
                  aria-haspopup="listbox"
                  aria-labelledby=":r14: :r13:"
                  class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                  id=":r13:"
                  role="combobox"
                  tabindex="0"
                >
                  10
                </div>
                <input
                  aria-hidden="true"
                  aria-invalid="false"
                  class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                  name=":r16:"
                  tabindex="-1"
                  value="10"
                />
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                  data-testid="ArrowDropDownIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </div>
              <p
                class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
              >
                1–1 of 1
              </p>
              <div
                class="MuiTablePagination-actions"
              >
                <button
                  aria-label="Go to previous page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to previous page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowLeftIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </button>
                <button
                  aria-label="Go to next page"
                  class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                  disabled=""
                  tabindex="-1"
                  title="Go to next page"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <main
      class="mainWrap"
      data-testid="users-page"
    >
      <div
        class="header"
      >
        <div
          class="rowOne"
        >
          <h3>
            Manajemen User
          </h3>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium backButton css-sghohy-MuiButtonBase-root-MuiButton-root"
            data-testid="back-button"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ArrowBackIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                />
              </svg>
            </span>
            Kembali
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="rowTwo"
        >
          <div
            class="MuiToggleButtonGroup-root MuiToggleButtonGroup-fullWidth toggleRole css-mgs850-MuiToggleButtonGroup-root"
            role="group"
          >
            <button
              aria-pressed="false"
              class="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU kRQhAE MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
              data-testid="button-verified-users"
              tabindex="0"
              type="button"
              value="false"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="VerifiedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m23 12-2.44-2.79.34-3.69-3.61-.82-1.89-3.2L12 2.96 8.6 1.5 6.71 4.69 3.1 5.5l.34 3.7L1 12l2.44 2.79-.34 3.7 3.61.82L8.6 22.5l3.4-1.47 3.4 1.46 1.89-3.19 3.61-.82-.34-3.69L23 12zm-12.91 4.72-3.8-3.81 1.48-1.48 2.32 2.33 5.85-5.87 1.48 1.48-7.33 7.35z"
                />
              </svg>
               
              Terverifikasi
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-pressed="true"
              class="MuiButtonBase-root MuiToggleButton-root Mui-selected MuiToggleButton-fullWidth MuiToggleButton-sizeSmall MuiToggleButton-standard sc-bdfCDU kRQhAE MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-1xgjc5y-MuiButtonBase-root-MuiToggleButton-root"
              data-testid="button-unverified-user"
              tabindex="0"
              type="button"
              value="true"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="UnpublishedIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M21.19 21.19 2.81 2.81 1.39 4.22l2.27 2.27C2.61 8.07 2 9.96 2 12c0 5.52 4.48 10 10 10 2.04 0 3.93-.61 5.51-1.66l2.27 2.27 1.41-1.42zm-10.6-4.59-4.24-4.24 1.41-1.41 2.83 2.83.18-.18 1.41 1.41-1.59 1.59zm3-5.84-7.1-7.1C8.07 2.61 9.96 2 12 2c5.52 0 10 4.48 10 10 0 2.04-.61 3.93-1.66 5.51L15 12.17l2.65-2.65-1.41-1.41-2.65 2.65z"
                />
              </svg>
               
              Tidak Terverifikasi
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <div
            class="searchInputWrap"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium iconSearch css-i4bv87-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
            <input
              class="searchInput"
            />
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root card css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiTableContainer-root css-41abqd-MuiTableContainer-root"
        >
          <table
            aria-label="sticky table"
            class="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          >
            <thead
              class="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              >
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead false css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Nama Lengkap
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead email css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Email
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead false css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Peran
                </th>
                <th
                  class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium tableHead css-y8ay40-MuiTableCell-root"
                  scope="col"
                >
                  Aksi
                </th>
              </tr>
            </thead>
            <tbody
              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            >
              <tr
                class="MuiTableRow-root MuiTableRow-hover css-1q1u3t4-MuiTableRow-root"
              >
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody false css-1ex1afd-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody email css-1ex1afd-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody false css-1ex1afd-MuiTableCell-root"
                />
                <td
                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium tableBody css-1ex1afd-MuiTableCell-root"
                >
                  <div
                    class="tableAction"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium buttonAction css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                      data-testid="button-delete"
                      tabindex="0"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="DeleteIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
                        />
                      </svg>
                      <div
                        class="email"
                      >
                        Hapus Akun
                      </div>
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
        <div
          class="MuiTablePagination-root tablePagination css-jtlhu6-MuiTablePagination-root"
        >
          <div
            class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular MuiTablePagination-toolbar css-78c6dr-MuiToolbar-root-MuiTablePagination-toolbar"
          >
            <div
              class="MuiTablePagination-spacer css-1psng7p-MuiTablePagination-spacer"
            />
            <p
              class="MuiTablePagination-selectLabel css-pdct74-MuiTablePagination-selectLabel"
              id=":r14:"
            >
              Rows per page:
            </p>
            <div
              class="MuiInputBase-root MuiInputBase-colorPrimary MuiTablePagination-input css-16c50h-MuiInputBase-root-MuiTablePagination-select"
            >
              <div
                aria-controls=":r15:"
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-labelledby=":r14: :r13:"
                class="MuiSelect-select MuiTablePagination-select MuiSelect-standard MuiInputBase-input css-194a1fa-MuiSelect-select-MuiInputBase-input"
                id=":r13:"
                role="combobox"
                tabindex="0"
              >
                10
              </div>
              <input
                aria-hidden="true"
                aria-invalid="false"
                class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                name=":r16:"
                tabindex="-1"
                value="10"
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiTablePagination-selectIcon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
                data-testid="ArrowDropDownIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <p
              class="MuiTablePagination-displayedRows css-levciy-MuiTablePagination-displayedRows"
            >
              1–1 of 1
            </p>
            <div
              class="MuiTablePagination-actions"
            >
              <button
                aria-label="Go to previous page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to previous page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowLeftIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </button>
              <button
                aria-label="Go to next page"
                class="MuiButtonBase-root Mui-disabled MuiIconButton-root Mui-disabled MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
                disabled=""
                tabindex="-1"
                title="Go to next page"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="KeyboardArrowRightIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
